### ISSUE TITLE
typecheck giving undefined error for existing field of struct

### ISSUE THREAD

**Issue by gharia** [golangci/golangci-lint#3668]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

I have below structs

```
type Config struct {
	*common.Config

	ElasticsearchURL string 
	Port             string       
}

// common.Config
type Config struct {
	Loader Loader

	AppName       string  
	AppBuild      string    
	LogLevel      string    
	Environment   util.Environment 
	Debug         bool
}
```
So `common.Config` is part of `Config`. Now wherever I use `Debug` from `Config`, typecheck gives the below lint error

```
c.Debug undefined (type *Config has no field or method Debug) (typecheck)
```

### Expected behaviour
It should not give a lint error as `Debug` is part of the struct (Or maybe I am missing something?)

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version v1.50.1 built from (unknown, mod sum: "h1:C829clMcZXEORakZlwpk7M4iDw2XiwxxKaG504SZ9zY=") on (unknown)
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
linters-settings:
  # report about assignment of errors to blank identifier: `num, _ := strconv.Atoi(numStr)`;
  check-blank: true

  cyclop:
    max-complexity: 15

  funlen:
    lines: 120
    statements: 100

  lll:
    # max line length, lines longer will be reported. Default is 120.
    # '\t' is counted as 1 character by default, and can be changed with the tab-width option
    line-length: 160
    # tab width in spaces. Default to 1.
    tab-width: 1

  govet:
    check-shadowing: true
    settings:
      shadow:
        strict: false

  golint:
    min-confidence: 0.5

  gocyclo:
    min-complexity: 20

  maligned:
    # print struct with more effective memory layout or not, false by default
    suggest-new: true

  dupl:
    threshold: 100

  goconst:
    # minimal length of string constant, 3 by default
    min-len: 3
    # minimal occurrences count to trigger, 3 by default
    min-occurrences: 3

  misspell:
    # Correct spellings using locale preferences for US or UK.
    # Default is to use a neutral variety of English.
    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
    locale: US

  nakedret:
    # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
    max-func-lines: 30

  goimports:
    # put imports beginning with prefix after 3rd-party packages;
    # it's a comma-separated list of prefixes
    local-prefixes: github.com/golangci/golangci-lint

  gocritic:
    enabled-tags:
      - performance
      - style
      - experimental
    disabled-checks:
      - wrapperFunc
      - whyNoLint

  tagliatelle:
    # Check the struck tag name case.
    case:
      # Use the struct field name to check the name of the struct tag.
      use-field-name: false # Default: false
      rules:
        # Any struct tag type can be used.
        # Support string case: `camel`, `pascal`, `kebab`, `snake`, `goCamel`, `goPascal`, `goKebab`, `goSnake`, `upper`, `lower`
        json: snake
        yaml: goCamel
        xml: snake
        bson: snake

linters:
  enable-all: true
  disable:
    - gochecknoglobals
    - lll
    - funlen
    - dupl
    - maligned
    - nolintlint
    - testpackage
    - nlreturn
    - exhaustive
    - gofumpt
    - exhaustivestruct
    - exhaustruct
    - goerr113
    - godot
    - gci
    - ireturn
    - nonamedreturns
    - varnamelen

issues:
  exclude-rules:
    - path: pkg/client/dataapi/client.go
      linters:
        - tagliatelle
    - path: pkg/models/calendar-api/
      linters:
        - tagliatelle
    - path: pkg/client/yodlee/
      linters:
        - tagliatelle
    - path: pkg/client/elasticsearch/
      linters:
        - tagliatelle
    - path: pkg/models/refdb/
      linters:
        - tagliatelle

run:
  deadline: 10m
  skip-dirs:
    - vendor
    - cmd/content-server-benchmark/
    - pkg/content/test

output:
  sort-results: true

severity:
  default-severity: error
  case-sensitive: false
  rules:
    - linters:
        - exhaustivestruct
      severity: warning

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.19.6 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/hemraj/.cache/go-build"
GOENV="/home/hemraj/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/dave/dev/go/pkg/mod"
GOOS="linux"
GOPATH="/home/dave/dev/go"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.19.6"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2764653345=/tmp/go-build -gno-record-gcc-switches"

```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 82 linters: [asasalint asciicheck bidichk bodyclose containedctx contextcheck cyclop deadcode decorder depguard dogsled dupword durationcheck errcheck errchkjson errname errorlint execinquery exportloopref forbidigo forcetypeassert gochecknoinits gocognit goconst gocritic gocyclo godox gofmt goheader goimports golint gomnd gomoddirectives gomodguard goprintffuncname gosec gosimple govet grouper ifshort importas ineffassign interfacebloat interfacer loggercheck maintidx makezero misspell nakedret nestif nilerr nilnil noctx nosnakecase nosprintfhostport paralleltest prealloc predeclared promlinter reassign revive rowserrcheck scopelint sqlclosecheck staticcheck structcheck stylecheck tagliatelle tenv testableexamples thelper tparallel typecheck unconvert unparam unused usestdlibvars varcheck wastedassign whitespace wrapcheck wsl] 
INFO [loader] Go packages loading at mode 575 (imports|name|deps|exports_file|files|types_sizes|compiled_files) took 1.25337667s 
WARN [runner] The linter 'nosnakecase' is deprecated (since v1.48.1) due to: The repository of the linter has been deprecated by the owner.  Replaced by revive(var-naming). 
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref. 
WARN [runner] The linter 'ifshort' is deprecated (since v1.48.0) due to: The repository of the linter has been deprecated by the owner.  
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused. 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 54.107745ms 
INFO [linters_context] importas settings found, but no aliases listed. List aliases under alias: key. 
INFO Memory: 885 samples, avg is 2377.7MB, max is 4129.5MB 
INFO Execution took 2m0.018179025s                
INFO [linters_context/goanalysis] analyzers took 25m45.009440746s with top 10 stages: gocritic: 13m34.788735809s, buildir: 4m25.988730307s, buildssa: 3m41.586936107s, bidichk: 39.227864315s, the_only_name: 23.582078036s, interfacer: 12.729865508s, inspect: 8.415070263s, unparam: 8.400580195s, unconvert: 7.280899926s, goimports: 5.44004505s 
WARN [linters_context] rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters_context] sqlclosecheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters_context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters_context] wastedassign is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649. 
INFO [runner/skip_dirs] Skipped 46 issues from dir pkg/content/test/get_data by pattern pkg/content/test 
INFO [runner/skip_dirs] Skipped 214 issues from dir pkg/content/test by pattern pkg/content/test 
INFO [runner/skip_dirs] Skipped 14 issues from dir pkg/content/test/elastic_import by pattern pkg/content/test 
WARN [runner/nolint] Found unknown linters in //nolint directives: errlint, gomnd fine to not using constant for retrier settings, gosec these are a constant, maligned - order makes more sense this way, maligned leave in logical order, not a secret., typecheck debug is already defined. is typecheck dumb?, unparam - for future use when gorm supports passing context 
INFO [runner/max_same_issues] 74/77 issues with text "type assertion must be checked" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 68/71 issues with text "assignments should only be cuddled with other assignments" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 64/67 issues with text "return statements should not be cuddled if block has more than two lines" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 37/40 issues with text "mnd: Magic number: 3, in <argument> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 29/32 issues with text "non-wrapping format verb for fmt.Errorf. Use `%w` to format errors" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 27/30 issues with text "if statements should only be cuddled with assignments" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 27/30 issues with text "json(snake): got '_id' want 'id'" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 24/27 issues with text "error returned from external package is unwrapped: sig: func encoding/json.Marshal(v any) ([]byte, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 23/26 issues with text "error returned from external package is unwrapped: sig: func encoding/json.Unmarshal(data []byte, v any) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 19/22 issues with text "only one cuddle assignment allowed before if statement" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 18/21 issues with text "bson(snake): got '_id' want 'id'" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 18/21 issues with text "mnd: Magic number: 3, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 16/19 issues with text "error returned from interface method should be wrapped: sig: func (github.com/labstack/echo/v4.Context).JSON(code int, i interface{}) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 16/19 issues with text "mnd: Magic number: 2, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 15/18 issues with text "expressions should not be cuddled with blocks" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 12/15 issues with text "mnd: Magic number: 4, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 11/14 issues with text "ioutilDeprecated: ioutil.ReadAll is deprecated, use io.ReadAll instead" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 10/13 issues with text "return both the `nil` error and invalid value: use a sentinel error instead" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 9/12 issues with text "Function TestPreset has missing the call to method parallel in the test run\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 9/12 issues with text "mnd: Magic number: 10.0, in <argument> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 9/12 issues with text "mnd: Magic number: 5, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 8/11 issues with text "error returned from external package is unwrapped: sig: func (*github.com/go-resty/resty/v2.Request).Get(url string) (*github.com/go-resty/resty/v2.Response, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 8/11 issues with text "error returned from external package is unwrapped: sig: func (*net/http.Client).Do(req *net/http.Request) (*net/http.Response, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 8/11 issues with text "error returned from external package is unwrapped: sig: func (*gopkg.in/mgo.v2.Query).All(result interface{}) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 8/11 issues with text "error returned from external package is unwrapped: sig: func time.Parse(layout string, value string) (time.Time, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 8/11 issues with text "mnd: Magic number: 5, in <argument> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 8/11 issues with text "SA1019: \"io/ioutil\" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. " were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 7/10 issues with text "Function TestBasic has missing the call to method parallel in the test run\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 6/9 issues with text "ranges should only be cuddled with assignments used in the iteration" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 6/9 issues with text "for statement without condition should never be cuddled" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 6/9 issues with text "only one cuddle assignment allowed before range statement" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 5/8 issues with text "error returned from external package is unwrapped: sig: func net/http.NewRequestWithContext(ctx context.Context, method string, url string, body io.Reader) (*net/http.Request, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 5/8 issues with text "error returned from external package is unwrapped: sig: func gitlab.benzinga.io/benzinga/bzgo/internal/sitemap-engine/sitemap.PublishSitemap(ctx context.Context, cfg *gitlab.benzinga.io/benzinga/bzgo/internal/sitemap-engine/config.Config, s3 *gitlab.benzinga.io/benzinga/bzgo/internal/sitemap-engine/storage/s3.Storage, sitemapFile string, data []byte) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 5/8 issues with text "mnd: Magic number: 2, in <condition> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 5/8 issues with text "error returned from external package is unwrapped: sig: func (*encoding/json.Decoder).Decode(v any) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 5/8 issues with text "mnd: Magic number: 10, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 4/7 issues with text "declarations should never be cuddled" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 4/7 issues with text "error returned from interface method should be wrapped: sig: func (github.com/labstack/echo/v4.Context).NoContent(code int) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 4/7 issues with text "mnd: Magic number: 2, in <argument> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 4/7 issues with text "mnd: Magic number: 3, in <condition> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 4/7 issues with text "error returned from external package is unwrapped: sig: func strconv.ParseFloat(s string, bitSize int) (float64, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "undeclared name: `log`" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "mnd: Magic number: 4, in <argument> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "mnd: Magic number: 1e6, in <operation> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "error returned from external package is unwrapped: sig: func gitlab.benzinga.io/benzinga/bzgo/pkg/util.ParseEnvironment(input string) (gitlab.benzinga.io/benzinga/bzgo/pkg/util.Environment, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "mnd: Magic number: 1e3, in <operation> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "only cuddled expressions if assigning variable or using from line above" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 3/6 issues with text "error returned from external package is unwrapped: sig: func github.com/cenk/backoff.RetryNotify(operation github.com/cenk/backoff.Operation, b github.com/cenk/backoff.BackOff, notify github.com/cenk/backoff.Notify) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "mnd: Magic number: 15, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "error returned from external package is unwrapped: sig: func (*github.com/uptrace/bun.InsertQuery).Exec(ctx context.Context, dest ...interface{}) (database/sql.Result, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "error returned from external package is unwrapped: sig: func (*github.com/uptrace/bun.DB).Exec(query string, args ...interface{}) (database/sql.Result, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "Function TestConfig missing the call to method parallel\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "Function TestClient has missing the call to method parallel in the test run\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "Function TestScreenerResult has missing the call to method parallel in the test run\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "mnd: Magic number: 10, in <argument> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "error returned from external package is unwrapped: sig: func (*github.com/uptrace/bun.SelectQuery).Scan(ctx context.Context, dest ...interface{}) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "branch statements should not be cuddled if block has more than two lines" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 2/5 issues with text "error returned from external package is unwrapped: sig: func io/ioutil.ReadAll(r io.Reader) ([]byte, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func (*github.com/uptrace/bun.DB).ScanRows(ctx context.Context, rows *database/sql.Rows, dest ...interface{}) error" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func (*github.com/olivere/elastic/v7.BoolQuery).Source() (interface{}, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func (*gopkg.in/mgo.v2.Collection).Upsert(selector interface{}, update interface{}) (info *gopkg.in/mgo.v2.ChangeInfo, err error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "mnd: Magic number: 30, in <assign> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func github.com/shopspring/decimal.NewFromString(value string) (github.com/shopspring/decimal.Decimal, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func (*github.com/uptrace/bun.DeleteQuery).Exec(ctx context.Context, dest ...interface{}) (database/sql.Result, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func gitlab.benzinga.io/benzinga/bzgo/pkg/util.LoadViperForApp(appName string) (*github.com/spf13/viper.Viper, error)" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "mnd: Magic number: 5, in <condition> detected" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "only one cuddle assignment allowed before switch statement" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "Function TestMetrics missing the call to method parallel\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "Function TestKeys has missing the call to method parallel in the test run\n" were hidden, use --max-same-issues 
INFO [runner/max_same_issues] 1/4 issues with text "error returned from external package is unwrapped: sig: func (*github.com/olivere/elastic/v7.SearchService).Do(ctx context.Context) (*github.com/olivere/elastic/v7.SearchResult, error)" were hidden, use --max-same-issues 
INFO [runner/max_from_linter] 228/278 issues from linter wrapcheck were hidden, use --max-issues-per-linter 
INFO [runner/max_from_linter] 110/160 issues from linter gomnd were hidden, use --max-issues-per-linter 
INFO [runner/max_from_linter] 75/125 issues from linter paralleltest were hidden, use --max-issues-per-linter 
INFO [runner/max_from_linter] 47/97 issues from linter tagliatelle were hidden, use --max-issues-per-linter 
INFO [runner/max_from_linter] 5/55 issues from linter wsl were hidden, use --max-issues-per-linter 
INFO [runner] Issues before processing: 19945, after processing: 420 
INFO [runner] Processors filtering stat (out/in): identifier_marker: 18014/18014, exclude: 18014/18014, uniq_by_line: 1598/11570, max_per_file_from_linter: 1598/1598, max_same_issues: 885/1598, sort_results: 420/420, autogenerated_exclude: 18014/19671, path_shortener: 420/420, severity-rules: 420/420, path_prefixer: 420/420, filename_unadjuster: 19945/19945, skip_files: 19945/19945, nolint: 11570/14574, max_from_linter: 420/885, source_code: 420/420, cgo: 19945/19945, path_prettifier: 19945/19945, skip_dirs: 19671/19945, exclude-rules: 14574/18014, diff: 1598/1598 
INFO [runner] processing took 1.805049252s with stages: exclude-rules: 1.009430576s, identifier_marker: 515.781055ms, nolint: 132.945373ms, path_prettifier: 65.319273ms, autogenerated_exclude: 56.534442ms, skip_dirs: 8.887229ms, cgo: 4.887498ms, source_code: 4.135662ms, filename_unadjuster: 2.893875ms, uniq_by_line: 2.275545ms, max_same_issues: 1.288275ms, sort_results: 221.309µs, max_from_linter: 154.274µs, path_shortener: 136.73µs, max_per_file_from_linter: 117.263µs, severity-rules: 38.708µs, skip_files: 804ns, exclude: 733ns, diff: 416ns, path_prefixer: 212ns 
INFO [runner] linters took 2m27.094988111s with stages: goanalysis_metalinter: 2m25.289771017s, rowserrcheck: 26.825µs, sqlclosecheck: 7.353µs, structcheck: 6.798µs, wastedassign: 5.922µs 
internal/content-server/config/config.go:147:7: c.Debug undefined (type *Config has no field or method Debug) (typecheck)
	if c.Debug {
	     ^

```

</details>

### Code example or link to a public repository

<details>

```go
// Struct added above
```

</details>

**Comment by ldez** [comment]
Hello,

`typecheck` is not a real linter it's just a way to parse/display "compilation" and linters errors (linter reports are not errors).

If you have a typecheck issue it's because you have a compilation error.

**Comment by gharia** [comment]
@ldez, that;s correct. But the thing is there are no compilation errors. The main `Config` has `*common.Config` and `common.Config` contains `Debug`. Code is running fine without issue

**Comment by ldez** [comment]
can you provide a full "working" reproducible example (not just a snippet)?

**Comment by natefinch** [comment]
@ldez this is happening all over GitHub's internal go repos. It's not a real compiler error. The code compiles just fine using the local go compiler. I suspect it's a version mismatch between the compiler used to build golangci-lint and the compiler used to build the code it's linting. I don't quite understand why that would matter, but it seems like, if the code is built with go 1.18 and it gets linted with a golangci-lint binary that was built with go 1.19, then the error comes up.... sometimes, but not always and not everywhere.  I'll try to get a simple repro scenario.

**Comment by filinvadim** [comment]
This must be reopen
